imgr.py

```diff
@@ -191,7 +197,7 @@ def upgrade(hutch, ioc, version):
     try:
         utils.COMMITHOST = vs["COMMITHOST"]
     except:
-        pass
+        pass # TODO(josh): no silent exceptions!
```

Comment:
COMMITHOST is optional.  If it's not there, this *should* fail silently.

Reply:
I guess I do not know enough (anything) about `COMMITHOST.` Generally I advise when you have are optionally populating a field warn / log if it doesn't get populated so if I future user builds a feature that then breaks off of it they at least have a breadcrumb log somewhere to fix their issue. 

May be totally off base for this specific case

Reply:
Feel free to rewrite this to test if "COMMITHOST" is a key of vs before doing the assignment, and get rid of the try.

Reply:
We can look for `KeyError` specifically to catch this